core/src/main/java/org/kiwix/kiwixmobile/core/search/viewmodel/State.kt

```diff
@@ -29,11 +29,26 @@ sealed class State {
   abstract val searchString: String
   abstract val searchOrigin: SearchOrigin
 
-  data class Results(
+  @Suppress("DataClassPrivateConstructor") // Make companion object act as constructor
```

Comment:
I would prefer if we weren't marking this as private because of the above warning.

I think an explicit named companion object function, I tend to use `from`, reads the best here. Operator overloading to make `Results.invoke(/*params*/)` look like a constructor is cool but violates the principle of least surprise.

It is that or mark `values` private and introduce `orderedValues` or some such field, potentially outside of the constructor

Sorry for going back and forth on this, just trying to puzzle out what expresses this best

Reply:
I changed the fake constructor to `from` as you requested.

I also changed the `private` to `@Deprecated` to make sure it is not accidentally called.

> mark values private and introduce orderedValues or some such field, potentially outside of the constructor

That might involve changing a lot of additional code.

Reply:
You know what you were right from the start, this should probably be in the SearchResultGenerator, it is being too awkward in the State. Really am sorry for changing my mind so much

Reply:
I moved it back as you requested